Ruby Release Notes (#19836)
newrelic/docs-website: src/content/docs/release-notes/agent-release-notes/ruby-release-notes/ruby-agent-9-17-0.mdx

- **Feature: Add Redshift as recognized ActiveRecord adapter**

  When the agent does not recognize an ActiveRecord adapter, the host, port, and database name information is not added to the datastore span. Redshift will now be treated like PostgreSQL, and the agent will save the host, port, and database name on the span. [PR#3032](https://github.com/newrelic/newrelic-ruby-agent/pull/3032)

- **Feature: Add instrumentation for aws-sdk-kinesis**

  The agent now has instrumentation for the [aws-sdk-kinesis](https://rubygems.org/gems/aws-sdk-kinesis) gem. It will record message broker segments for `get_records`, `put_record`, and `put_records` operations. All other operations will record standard segments. [PR#2974](https://github.com/newrelic/newrelic-ruby-agent/pull/2974)

- **Bugfix: Stop emitting inaccurate debug-level log about deprecated configuration options**

>   In the previous major release, we dropped support for `disable_<library_name>` configuration options in favor of `instrumentation.<library_name>`. Previously, a DEBUG level log warning appeared whenever `disable_*` options were set to `true`, even for libraries (e.g. Action Dispatch) without equivalent `instrumentation.*` options:

Reviewer: ```suggestion
  In the previous major release, we dropped support for many `disable_library_name` configuration options in favor of `instrumentation.library_name`. Previously, a DEBUG-level log warning appeared whenever `disable_*` options were set to `true`, even for libraries (e.g. Action Dispatch) without equivalent `instrumentation.*` options:
```

>   >DEBUG : [DEPRECATED] configuration disable_<library_name> for <library_name> will be removed in the next major release. Use instrumentation.<library_name> with one of ["auto", "disabled", "prepend", "chain"]

Reviewer: ```suggestion
  >DEBUG : [DEPRECATED] configuration disable_action_dispatch for Action Dispatch will be removed in the next major release. Use instrumentation.action_dispatch with one of ["auto", "disabled", "prepend", "chain"]
```

>   This inaccurate warning has been removed. If you are disabling instrumentation using `instrumentation.<library_name>: disabled` or `NEW_RELIC_INSTRUMENTATION_<LIBRARY_NAME>=disabled`, please verify the option exists by consulting our [configuration documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation). If the option does not exist, check the ['Disabling' section](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#disabling) to see if there is a related option. We apologize for the confusion. [PR#3005](https://github.com/newrelic/newrelic-ruby-agent/pull/3005)

Reviewer: ```suggestion
  This inaccurate warning has been removed. If you are disabling instrumentation using `instrumentation.library_name: disabled` or `NEW_RELIC_INSTRUMENTATION_LIBRARY_NAME=disabled`, please verify the option exists by consulting our [configuration documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation). If the option does not exist, check the ['Disabling' section](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#disabling) to see if there is a related option. We apologize for the confusion. [PR#3005](https://github.com/newrelic/newrelic-ruby-agent/pull/3005)
```
